src/libostree/ostree-sysroot-deploy.c

```diff
@@ -874,114 +875,110 @@ ostree_sysroot_write_origin_file (OstreeSysroot         *sysroot,
 }
 
 static gboolean
-get_kernel_from_tree (GFile         *deployroot,
-                      GFile        **out_kernel,
-                      GFile        **out_initramfs,
-                      GCancellable  *cancellable,
-                      GError       **error)
+get_kernel_from_tree (int             deployment_dfd,
+                      int            *out_boot_dfd,
+                      char          **out_kernel_name,
+                      char          **out_initramfs_name,
+                      GCancellable   *cancellable,
+                      GError        **error)
 {
   gboolean ret = FALSE;
-  gs_unref_object GFile *ostree_bootdir
-    = g_file_resolve_relative_path (deployroot, "usr/lib/ostree-boot");
-  gs_unref_object GFile *bootdir = g_file_get_child (deployroot, "boot");
-  gs_unref_object GFileEnumerator *dir_enum = NULL;
-  gs_unref_object GFile *ret_kernel = NULL;
-  gs_unref_object GFile *ret_initramfs = NULL;
+  glnx_fd_close int ret_boot_dfd = -1;
+  g_auto(GLnxDirFdIterator) dfditer = { 0, };
+  g_autofree char *ret_kernel_name = NULL;
+  g_autofree char *ret_initramfs_name = NULL;
   gs_free char *kernel_checksum = NULL;
   gs_free char *initramfs_checksum = NULL;
 
-  if (g_file_query_exists (ostree_bootdir, NULL))
+  ret_boot_dfd = glnx_opendirat_with_errno (deployment_dfd, "usr/lib/ostree-boot", TRUE);
+  if (ret_boot_dfd == -1)
     {
-      dir_enum = g_file_enumerate_children (ostree_bootdir, OSTREE_GIO_FAST_QUERYINFO,
-                                            G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-                                            NULL, error);
-      if (!dir_enum)
-        goto out;
-    }
-  else
-    {
-      dir_enum = g_file_enumerate_children (bootdir, OSTREE_GIO_FAST_QUERYINFO,
-                                            G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-                                            NULL, error);
-      if (!dir_enum)
-        goto out;
+      if (errno != ENOENT)
+        {
+          glnx_set_error_from_errno (error);
```

Comment:
Maybe use `glnx_set_prefix_error_from_errno()` here to match the behavior of `glnx_opendirat()`?